realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -404,6 +405,18 @@ public void equals_cyclicObject() {
     }
 
     @Test
+    public void equals_customMethod() {
```

Comment:
I would also add a test that compared equals for a un-managed and managed object.